File: server/services/accounts.js
import { API_CONFIG } from '../config/api.js';
import { getAccessToken } from './auth.js';

export async function getAccountDetails(accountId) {
  const token = await getAccessToken();
  const response = await fetch(`${API_CONFIG.baseUrl}/accounts/${accountId}/`, {
    headers: {
      ...API_CONFIG.headers,
      Authorization: `Bearer ${token}`,
    },
  });

  if (!response.ok) {
    throw new Error(`Failed to fetch account details: ${response.status}`);
  }

  return response.json();
}

File: server/src/services/accounts.py
import httpx

from .auth import API_CONFIG, get_access_token


async def get_account_details(account_id: str):
    token = get_access_token()

    async with httpx.AsyncClient() as client:
        response = await client.get(
            f"{API_CONFIG['base_url']}/accounts/{account_id}/",
            headers={**API_CONFIG["headers"], "Authorization": f"Bearer {token}"},
        )
        response.raise_for_status()
        return response.json()
